sbws/lib/v3bwfile.py

```diff
@@ -544,29 +550,56 @@ def from_results(cls, results, secs_recent=None, secs_away=None,
         # In #28563 will include again these lines, but make them unparseable
         # by Tor.
         # This might confirm #28042.
+        exclusion_reason = None
         kwargs['relay_recent_measurement_exclusion_not_success_count'] = \
             len(results) - len(success_results)
         if not success_results:
-            return None, 'recent_measurement_exclusion_not_success_count'
+            exclusion_reason = 'recent_measurement_exclusion_not_success_count'
         results_away = \
             cls.results_away_each_other(success_results, secs_away)
         kwargs['relay_recent_measurement_exclusion_not_distanciated_count'] = \
             len(success_results) - len(results_away)
         if not results_away:
-            return None, 'recent_measurement_exclusion_not_distanciated_count'
+            exclusion_reason = \
+                'recent_measurement_exclusion_not_distanciated_count'
         # log.debug("Results away from each other: %s",
         #           [unixts_to_isodt_str(r.time) for r in results_away])
         results_recent = cls.results_recent_than(results_away, secs_recent)
         kwargs['relay_recent_measurement_exclusion_not_recent_count'] = \
             len(results_away) - len(results_recent)
         if not results_recent:
-            return None, 'recent_measurement_exclusion_not_recent_count'
+            exclusion_reason = \
+                'recent_measurement_exclusion_not_recent_count'
 
         if not len(results_recent) >= min_num:
             kwargs['relay_recent_measurement_exclusion_not_min_num_count'] = \
                 len(results_recent)
             # log.debug('The number of results is less than %s', min_num)
-            return None, 'recent_measurement_exclusion_not_min_num_count'
+            exclusion_reason = \
+                'recent_measurement_exclusion_not_min_num_count'
+
+        # Create a bandwidth line with the relay, but set ``vote=0`` so that
+        # Tor versions with patch #29806 does not vote on the relay.
+        # Set ``bw=0`` so that Tor versions without the patch,
+        # will give the relay low bandwidth.
+        # Include ``unmeasured=1`` in case Tor would vote on unmeasured relays
+        # in future versions.
+        kwargs['vote'] = '0'
```

Comment:
It might make more sense to only set these if we have an exclusion_reason.

Reply:
Changed